R/generic_import.R

```diff
@@ -0,0 +1,162 @@
+#' Generic import object.
+#'
+#' This shouldn't be initialised directly - this object manages transactions
+#' and the db connection and should be extended to implement actual imports.
+Import <- R6::R6Class(
```

Comment:
Decided to stick with the super class here over composition because I thought it does fit into the inheritance pattern but would be interested to get your opinion on that too? We can always separate it later if this starts becoming a bit of a god a class

Reply:
I am happy either way, really. It's fairly internally facing so we're free to move to composition later if we want